src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarHeader.Write.cs

```diff
@@ -819,5 +832,27 @@ private static string GenerateGlobalExtendedAttributeName(int globalExtendedAttr
 
             return result;
         }
+
+        // Returns the text's utf16 length truncated at the specified max length.
+        private static int GetUtf16TruncatedTextLength(ReadOnlySpan<char> text, int maxLength)
+        {
```

Comment:
Since presumably all ASCII and names less than maxLength are very common (yes?), consider adding:
```C#
int encodedLength = Encoding.UTF8.GetByteCount(text);
if (encodedLength <= maxLength)
{
    return text.Length;
}

```
at the beginning of the method.  That will be much faster than enumerating the runes and _should_ handle the majority cases.

Reply:
@Jozkee I don't see this being addressed? Unresolving conversation.

Reply:
I moved the `UTF8.GetByteCount(text)` to the callsites as I needed to know the result for a few more checks. I can bring the check back as a debug.assert.

Reply:
No, it's fine if you need it for more checks. I see you added the assert.